### PR TITLE
OGD-387 Employments by NINO endpoint - Sandbox + Documentation

### PR DESCRIPTION
**Pull request by edinhodzic**

### CODE REVIEW

File: app/uk/gov/hmrc/individualsapi/config/microserviceGlobal.scala [hunk 1]
```diff
@@ -93,8 +94,10 @@ object MicroserviceGlobal extends DefaultMicroserviceGlobal with ServiceLocatorR
   override def onRouteRequest(originalRequest: RequestHeader): Option[Handler] = {
     val requestContext = extractUriContext(originalRequest)
     if (unversionedContexts.contains(requestContext)) {
+      System.out.println(s"########## original [$originalRequest]")
```
Comment (edgarjimenez) [inline app/uk/gov/hmrc/individualsapi/config/microserviceGlobal.scala]: this is probably meant to be removed.

File: test/component/uk/gov/hmrc/individualsapi/controllers/HttpOptionsControllerV2Spec.scala [hunk 2]
```diff
@@ -63,7 +63,7 @@ class HttpOptionsControllerV2Spec extends BaseSpec {
 
     scenario("valid OPTIONS request to the sandbox implementation") {
 
-      When("I request citizen's details resource OPTIONS")
+      When("I request income resource OPTIONS")
```
Comment (edgarjimenez) [inline test/component/uk/gov/hmrc/individualsapi/controllers/HttpOptionsControllerV2Spec.scala]: good catch

File: test/unit/uk/gov/hmrc/individualsapi/services/EmploymentsServiceSpec.scala [hunk 3]
```diff
@@ -101,13 +101,20 @@ class EmploymentsServiceSpec extends UnitSpec with MockitoSugar with Dates {
...
-      val jan2016ToMar2017Interval = toInterval("2016-01-01", "2017-03-01")
+    val jan2016ToMar2017Interval = toInterval("2016-01-01", "2017-03-01")
+
+    "throw a matching exception when a nino match is unsuccessful based on a match id" in new Setup {
       a[MatchNotFoundException] should be thrownBy {
         await(sandboxEmploymentsService.paye(UUID.randomUUID(), jan2016ToMar2017Interval))
       }
     }
 
+    "throw a matching exception when a nino match is unsuccessful based on a nino" in new Setup {
```
Comment (edgarjimenez) [inline test/unit/uk/gov/hmrc/individualsapi/services/EmploymentsServiceSpec.scala]: We should test the same scenarios we test for matchId, such as data included for a particular date range and/or sorting.
I realise we basically use the same code to generate both lists, but that could easily be refactored at some point in the future and we would not have any tests to detect a problem.

File: app/uk/gov/hmrc/individualsapi/play/RequestHeaderUtils.scala [hunk 4]
```diff
@@ -37,7 +37,7 @@ object RequestHeaderUtils {
     )
   }
 
-  private def getVersion(originalRequest: RequestHeader) =
+  def getVersion(originalRequest: RequestHeader) =
```
Comment (tom59) [inline app/uk/gov/hmrc/individualsapi/play/RequestHeaderUtils.scala]: why changing it to public?

Comment (edinhodzic) [inline app/uk/gov/hmrc/individualsapi/play/RequestHeaderUtils.scala]: accidental commit

File: app/uk/gov/hmrc/individualsapi/config/microserviceGlobal.scala (hunk 1 above)
Comment (edinhodzic) [inline app/uk/gov/hmrc/individualsapi/config/microserviceGlobal.scala]: yeah, accidental commit

**Pull request merged**